tests/drum/test_other_cases.py

```diff
@@ -239,37 +241,63 @@ def test_set_language(
             assert case
 
     @pytest.mark.parametrize(
-        "framework, language", [(SKLEARN, PYTHON_ALL_HOOKS), (RDS, R_ALL_HOOKS)]
+        "framework, language, hooks_list, target_type",
+        [
+            (
+                None,
+                PYTHON_ALL_PREDICT_STRUCTURED_HOOKS,
+                CustomHooks.ALL_PREDICT_STRUCTURED,
+                REGRESSION,
+            ),
+            (None, R_ALL_PREDICT_STRUCTURED_HOOKS, CustomHooks.ALL_PREDICT_STRUCTURED, REGRESSION),
+            (
+                None,
+                PYTHON_ALL_PREDICT_UNSTRUCTURED_HOOKS,
+                CustomHooks.ALL_PREDICT_UNSTRUCTURED,
+                UNSTRUCTURED,
+            ),
+            (
+                None,
+                R_ALL_PREDICT_UNSTRUCTURED_HOOKS,
+                CustomHooks.ALL_PREDICT_UNSTRUCTURED,
+                UNSTRUCTURED,
+            ),
+        ],
     )
-    def test_custom_model_with_all_predict_hooks(
+    def test_custom_model_with_all_hooks(
         self,
         resources,
         framework,
         language,
+        hooks_list,
+        target_type,
         tmp_path,
     ):
         custom_model_dir = _create_custom_model_dir(
             resources,
             tmp_path,
             framework,
-            REGRESSION,
+            None,
             language,
         )
 
         input_dataset = resources.datasets(framework, REGRESSION)
 
         output = tmp_path / "output"
 
-        cmd = "{} score --code-dir {} --input {} --output {} --target-type regression".format(
-            ArgumentsOptions.MAIN_COMMAND, custom_model_dir, input_dataset, output
+        cmd = "{} score --code-dir {} --input {} --output {} --target-type {}".format(
+            ArgumentsOptions.MAIN_COMMAND, custom_model_dir, input_dataset, output, target_type
         )
         _exec_shell_cmd(
             cmd, "Failed in {} command line! {}".format(ArgumentsOptions.MAIN_COMMAND, cmd)
         )
-        preds = pd.read_csv(output)
-        assert all(
-            val for val in (preds["Predictions"] == len(CustomHooks.ALL_PREDICT)).values
-        ), preds
+        if hooks_list == CustomHooks.ALL_PREDICT_STRUCTURED:
+            preds = pd.read_csv(output)
+            assert all(val for val in (preds["Predictions"] == len(hooks_list)).values), preds
+        elif hooks_list == CustomHooks.ALL_PREDICT_UNSTRUCTURED:
+            with open(output) as f:
```

Comment:
this uses a default encoding which is platform dependant, which is utf-8.

Reply:
The test saves this file in utf8.